### PR TITLE
add MutableTree.SetInitialVersion()

### PR DESCRIPTION
**Pull request by erikgrinaker**
Requested in https://github.com/cosmos/cosmos-sdk/pull/7089#issuecomment-680877202.

I didn't add a `GetInitialVersion()`, since it's not clear whether this should return only the option or the _actual_ initial version that exists in a tree.

It might be cleaner to add `GetOptions()`, `SetOptions()`, and `UpdateOptions(func (o Options) Options)`, but we may not want all options to be mutable after construction.

CC @amaurymartiny

### CODE REVIEW

**Review by amaurymartiny — APPROVED** [review]
lgtm, I believe this will unblock my PR on the cosmos-sdk 👍

**Comment by erikgrinaker** [comment]
I'll need an approval from a grown-up, I'm afraid - the repo is set up like that. @alexanderbez, want to do the honors?

**Pull request merged**
